### ISSUE TITLE
opt: lookup join with no equality columns

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#41676]
https://teamcity.cockroachdb.com//viewLog.html?buildId=1542474&buildTypeId=Cockroach_SqlLogicTests

Easy repro:
```
CREATE TABLE t(pk INT PRIMARY KEY, col0 INT, col1 INT, col2 INT, col4 INT, UNIQUE INDEX (col2));
SELECT pk FROM t WHERE col4 = 1 AND col0 = 1 AND col2 IN (SELECT col0 FROM t WHERE col0 = 1 AND col2 IS NULL);
pq: index join must be against primary index
```
Possibly present on 19.2 branch.

**Comment by RaduBerinde** [comment]
Can repro on release-19.2.

**Comment by RaduBerinde** [comment]
Fixed in #41685.
